examples/audio/eeg_signal.py

```diff
@@ -0,0 +1,554 @@
+
+Gdown is an external package used to download large files from Google Drive. To know
+more, you can refer to its [PyPi page here](https://pypi.org/project/gdown)
+"""
+
+"""
+## Setup and Import headers
+"""
+
+"""shell
+pip install gdown
+pip install gdown
+pip install sklearn
+pip install pandas
+pip install numpy
+pip install matplotlib
+gdown 1V5B7Bt6aJm0UHbR7cRKBEK8jx7lYPVuX
+# gdown will download eeg-data.csv onto the local drive for use. Total size of
+eeg-data.csv is 105.7 MB
+"""
+
+import pandas as pd
+import matplotlib.pyplot as plt
+import json
+import numpy as np
+from tensorflow import keras
+from tensorflow.keras import layers
+import tensorflow as tf
+from sklearn import preprocessing, model_selection
+import random
+
+QUALITY_THRESHOLD = 128
+BATCH_SIZE = 64
+SHUFFLE_BUFFER_SIZE = 100
+
+"""
+## Read data from `eeg-data.csv`
+
+We use the Pandas library to read the `eeg-data.csv` file and display the first 5 rows
+using the `.head()` command
+"""
+
+eeg = pd.read_csv("eeg-data.csv")
+
+"""
+We remove unlabeled samples from our dataset as they do not contribute to the model. We
+also perform a `.drop()` operation on the columns that are not required for training data
+preparation
+"""
+
+unlabeled_eeg = eeg[eeg["label"] == "unlabeled"]
+eeg = eeg.loc[eeg["label"] != "unlabeled"]
+eeg = eeg.loc[eeg["label"] != "everyone paired"]
+
+eeg.drop(
+    [
+        "indra_time",
+        "Unnamed: 0",
+        "browser_latency",
+        "reading_time",
+        "attention_esense",
+        "meditation_esense",
+        "updatedAt",
+        "createdAt",
+    ],
+    axis=1,
+    inplace=True,
+)
+
+eeg.reset_index(drop=True, inplace=True)
+eeg.head()
+
+"""
+In the data, the samples recorded are given a score from 0 to 128 based on how
+well-calibrated the sensor was (0 being best, 200 being worst). We filter the values
+based on an arbitrary cutoff limit of 128.
+"""
+
+
+def makeNumpy(value_string):
+    str_list = json.loads(value_string)
+    return str_list
+
+
+eeg["raw_values"] = eeg["raw_values"].apply(makeNumpy)
+
+eeg = eeg.loc[eeg["signal_quality"] < QUALITY_THRESHOLD]
+print(eeg.shape)
+eeg.head()
+
+"""
+## Visualize one random sample from the data
+"""
+
+"""
+We visualize one sample from the data to understand how the stimulus-induced signal looks
+like
+"""
+
+
+def view_eeg_plot(idx):
+    data = eeg.loc[idx, "raw_values"]
+    plt.plot(data)
+    plt.title(f"Sample random plot")
+    plt.show()
+
+
+view_eeg_plot(7)
+
+"""
+## Pre-process and collate data
+"""
+
+"""
+There are a total of 67 different labels present in the data, where there are numbered
+sub-labels. We collate them under a single label as per their numbering and replace them
+in the data itself. Following this process, we perform simple Label encoding to get them
+in an integer format.
+"""
+
+print("Before replacing labels")
+print(eeg["label"].unique(), "\n")
+print(len(eeg["label"].unique()), "\n")
+
+
+def updateLabel(data):
+    data.replace(
```

Comment:
given that this is all in-place  and only called once we can also just drop the method entirely

Reply:
Alright, done